src/main/java/com/atlassian/jira/cloud/jenkins/buildinfo/service/JiraBuildInfoSenderImpl.java

```diff
@@ -119,14 +121,20 @@ public JiraSendInfoResponse sendBuildInfo(final JiraBuildInfoRequest request) {
     }
 
     private Set<String> getIssueKeys(final JiraBuildInfoRequest request, final WorkflowRun build) {
-        return Optional.ofNullable(request.getBranch())
-                .map(
-                        branch ->
-                                IssueKeyStringExtractor.extractIssueKeys(branch)
-                                        .stream()
-                                        .map(IssueKey::toString)
-                                        .collect(Collectors.toSet()))
-                .orElseGet(() -> issueKeyExtractor.extractIssueKeys(build));
+        Set<String> branchIssueKeys =
+                Optional.ofNullable(request.getBranch())
+                        .map(
+                                branch ->
+                                        IssueKeyStringExtractor.extractIssueKeys(branch)
+                                                .stream()
+                                                .map(IssueKey::toString)
+                                                .collect(Collectors.toSet()))
+                        .orElseGet(() -> issueKeyExtractor.extractIssueKeys(build));
+        Set<String> CommitIssueKeys = changeLogIssueKeyExtractor.extractIssueKeys(build);
+        if (!CommitIssueKeys.isEmpty()) {
+            branchIssueKeys.addAll(CommitIssueKeys);
+        }
+        return branchIssueKeys;
```

Comment:
If I understand this correctly, we are now including issue keys from **both the commit messages and the branch name**?

Reply:
Yes. Correct.